packages/microwave/src/lib/microwave.ts

```diff
@@ -0,0 +1,160 @@
+ * @deprecated 🚧 Work in progress.
+ */
+export function Microwave() {
+  return function MicrowaveDecorator<T>(originalClass: Type<T>): Type<T> {
+    const microwavedProxy = _decorateClass(
+      originalClass as Type<Microwaved<T>>,
+      {
+        wrapFactory(factoryFn) {
+          return _microwave(factoryFn);
+        },
+        preSet(target, property, value: unknown) {
+          const subject = _getOrCreateSubject(target, property);
+          // @todo fix this
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          subject.next(value as any);
+          target[_MARK_FOR_CHECK_SUBJECT_SYMBOL]?.next();
+        },
+      }
+    );
+
+    /* Copy static properties. */
+    Object.assign(microwavedProxy, originalClass);
+
+    return microwavedProxy;
+  };
+}
+
+/**
+ * @deprecated 🚧 Work in progress.
+ */
+export function watch<T, K extends keyof T = keyof T>(
+  component: T,
+  property: K
+): Observable<T[K]> {
+  if (!isMicrowaved(component)) {
+    throw new Error(
+      `Component is not microwaved. Did you add @Microwave decorator?`
+    );
+  }
+
+  return _getOrCreateSubject(component, property).pipe(
+    distinctUntilChanged(),
+    takeUntil(component[_DESTROYED_SUBJECT_SYMBOL])
+  );
+}
+
+export function isMicrowaved<T>(component: T): component is Microwaved<T> {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return (component as any)[_MARK_FOR_CHECK_SUBJECT_SYMBOL] != null;
+}
+
+export function _getOrCreateSubject<T, K extends keyof T = keyof T>(
+  component: T,
+  property: K
+): BehaviorSubject<T[K]> {
+  const microwaved = component as Microwaved<T>;
+  const subjects = microwaved[_SUBJECTS_SYMBOL];
+
+  let subject = subjects?.get(property);
+  if (subject == null) {
+    /* Use value from component if it is initialized in constructor. */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    subject = new BehaviorSubject(component[property] as any);
+    subjects?.set(property, subject);
+  }
+
+  // @todo fix this generic issue.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return subject as any;
+}
+
+/**
+ * Override component factory and trigger change detection.
+ */
+export function _microwave<T>(factoryFn: () => Microwaved<T>) {
+  /* A subject that regroups change detection requests
+   * so we can coalesce and trigger change detection
+   * with a custom strategy. */
+  const markForCheck$ = new Subject<void>();
+
+  /* Grab change detector to control it. */
+  const cdr = ɵɵdirectiveInject(ChangeDetectorRef);
+  cdr.detach();
+
+  const target = factoryFn();
+
+  const destroyed$ = new ReplaySubject<void>(1);
+  target[_MARK_FOR_CHECK_SUBJECT_SYMBOL] = markForCheck$;
+  target[_SUBJECTS_SYMBOL] = new Map();
+  target[_DESTROYED_SUBJECT_SYMBOL] = destroyed$;
+
+  markForCheck$
+    .pipe(
+      debounce(() => Promise.resolve()),
+      takeUntil(destroyed$)
+    )
+    .subscribe(() => cdr.detectChanges());
```

Comment:
As an evolution, it would be great to abstract rendering work to be able to pass custom rendering strategies (like the ones from rxa)

Reply:
That's the exact goal!
I am thinking about something like `@Microwave({strategy: ...})` parameter.
We'll have to discuss the interface of the strategy and other parameters: coalescing, renderScheduler etc...